### PR TITLE
Add Adobe Target Web Destination

### PR DESCRIPTION
**Pull request by marinhero**
This PR adds support for Adobe Target Web Mode. The capabilities added to the new destination are:

- Loading the SDK. Bring in the Adobe Target SDK into the current web app.
- Placeholder action: `upsertProfile`.

## Testing

Tested completed successfully via `yarn dev` test app.

Here are the settings & subscription I used to test:

```
{
  "client_code": "segmentexchangepartn",
  "admin_number": "10",
  "version": "2.8.0",
  "cookie_domain": "localhost",
  "mbox_name": "target-global-mbox",
  "subscriptions": [
    {
      "partnerAction": "upsertProfile",
      "name": "Upsert Profile",
      "enabled": true,
      "subscribe": "type = \"identify\"",
      "mapping": {}
    }
  ]
}
```

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [ ] [Segmenters] Tested in the staging environment

### CODE REVIEW

File: packages/browser-destinations/src/destinations/adobe-target/index.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+export const destination: BrowserDestinationDefinition<Settings, unknown> = {
+  name: 'Adobe Target Web',
+  slug: 'actions-adobe-target-web',
+  mode: 'device',
+
+  settings: {
+    client_code: {
+      label: 'Client Code',
+      description:
+        'Your client code is available at the top of the Administration > Implementation page of the Target interface.',
```
Comment (kdaswani) [inline packages/browser-destinations/src/destinations/adobe-target/index.ts:24]:
```suggestion
        'Your Adobe Target client code. To find your client code in Adobe Target, navigate to Administration > Implementation. The client code is shown at the top under Account Details.',
```

File: packages/browser-destinations/src/destinations/adobe-target/index.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        {
+          value: '2.8.0',
+          label: '2.8.0'
+        }
+      ],
+      default: '2.8.0',
+      required: true
+    },
+    mbox_name: {
+      label: 'MBox Name',
```
Comment (kdaswani) [inline packages/browser-destinations/src/destinations/adobe-target/index.ts:49]:
```suggestion
      label: 'Mbox Name',
```

File: packages/browser-destinations/src/destinations/adobe-target/index.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+      description: 'The name of the mbox to use',
```
Comment (kdaswani) [inline packages/browser-destinations/src/destinations/adobe-target/index.ts:50]:
```suggestion
      description: 'The name of the Adobe Target mbox to use. Defaults to `target-global-mbox`.',
```

File: packages/browser-destinations/src/destinations/adobe-target/index.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+      type: 'string',
+      required: true,
+      default: 'target-global-mbox'
+    },
+    cookie_domain: {
+      label: 'Cookie Domain',
+      description: 'The domain of the platform that your integration will run on',
```
Comment (kdaswani) [inline packages/browser-destinations/src/destinations/adobe-target/index.ts:57]:
```suggestion
      description: 'The domain from which you serve the mbox. Adobe Target recommends setting this value to your company's top-level domain.',
```

File: packages/browser-destinations/src/destinations/adobe-target/upsertProfile/index.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import type { BrowserActionDefinition } from '../../../lib/browser-destinations'
+import type { Settings } from '../generated-types'
+import { Adobe } from '../types'
+import type { Payload } from './generated-types'
+
+const action: BrowserActionDefinition<Settings, Adobe, Payload> = {
+  title: 'Upsert Profile',
+  description: 'Update or Create a user profile in Adobe Target',
```
Comment (kdaswani) [inline packages/browser-destinations/src/destinations/adobe-target/upsertProfile/index.ts:8]:
```suggestion
  description: 'Create or update a user profile in Adobe Target.',
```

File: packages/browser-destinations/src/destinations/adobe-target/index.ts [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+      label: 'Mbox Name',
+      description: 'The name of the Adobe Target mbox to use. Defaults to `target-global-mbox`.',
+      type: 'string',
+      required: true,
+      default: 'target-global-mbox'
+    },
+    cookie_domain: {
+      label: 'Cookie Domain',
+      description:
+        "The domain from which you serve the mbox. Adobe Target recommends setting this value to your company's top- level domain.",
```
Comment (kdaswani) [inline packages/browser-destinations/src/destinations/adobe-target/index.ts:58]:
```suggestion
        'The domain from which you serve the mbox. Adobe Target recommends setting this value to your company's top-level domain.',
```

File: packages/browser-destinations/src/destinations/adobe-target/upsertProfile/index.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+const action: BrowserActionDefinition<Settings, Adobe, Payload> = {
+  title: 'Upsert Profile',
+  description: 'Create or update a user profile in Adobe Target.',
+  platform: 'web',
+  defaultSubscription: 'type = "identify"',
+  fields: {
+    profile: {
+      type: 'object',
+      required: false,
```
Comment (nick-Ag) [inline packages/browser-destinations/src/destinations/adobe-target/upsertProfile/index.ts:14]: Can delete this line

**Review by nick-Ag — APPROVED** [review]
Tested and lgtm!

**Pull request merged**